packages/components/link/src/link.visualroute.jsx

```diff
@@ -1,17 +1,12 @@
 import { Link } from '@commercetools-frontend/ui-kit';
-import { ThemeProvider } from '@emotion/react';
+import { ThemeProvider } from '@commercetools-uikit/design-system';
 import { Suite, Spec } from '../../../../test/percy';
 
 export const routePath = '/link';
 
 const intlMessage = { id: 'link', defaultMessage: 'Link' };
 
-const purpleTheme = {
-  colorPrimary: 'purple',
-  colorPrimary25: 'deeppurple',
-};
```

Comment:
So my point of view is that it does not make sense anymore to have theming tests for specific components so I would remove them.
Theming is global and I guess what we would need to test is just that the `ThemeProvider` works when you provide and allowed theme name (and maybe also the `changeName` helper).

There's a chance we also would want to have VRTs for every component in every theme so we make sure they look like they're supposed to. I wouldn't implement that, at least for now. And, if we ever do, perhaps we should look into some way to automate it instead of writing different theming tests for every component.

Also, I would not implement (at least for now) the ability for the users to provide a custom theme, ui-kit is tailored for Merchant Center and we want to control how it looks like, so themes should be controlled by us.

Reply:
FYI: In the docs-kit we do use some of the UI Kit components and we've also been using the theme to override some of the UI Kit design tokens. For example:

https://github.com/commercetools/commercetools-docs-kit/blob/2d3bf6efc3c4768cc3056d47da9ecb3fe21a77ae/packages/ui-kit/src/design-system.ts#L303-L306

Something to keep in mind maybe. We don't have to keep the same functionality (and it's fine if we need to refactor/change some stuff in docs-kit) but the end result should be the same ideally.

Reply:
It seems then we would need to implement the functionality to allow users to overwrite a theme since what they want (in `docs-kit`) is to change the font used by `ui-kit` components. Maybe they are using `emotion` theming internally since they are initializing the provider with `ui-kit` properties but also another one which does not belong to `ui-kit` design tokens ([reference](https://github.com/commercetools/commercetools-docs-kit/blob/3610dc6f692d281f1a1ad37653aed2134b40175e/packages/gatsby-theme-docs/src/layouts/internals/layout-application.js#L46)).

Reply:
Thanks for the comments 🙏 and for bringing up the `docs-kit`. I will look for the best option to keep vrts passing and provide backwards compatibility in the `docs-kit`

Reply:
The docs-kit has its own "design system" tokens (partly referencing UI Kit design tokens) and its own version of "theming" (using Emotion theming yes).

However, I would be happy and ok to refactor things there to follow the same "theming setup" that we plan to use here.

Let's sync on that next week maybe, but I wouldn't worry much about it.

Reply:
What I did for now was adding 3 additional themes (apart from `vrtDark`): `vrtLink`, `vrtTag` and `vrtTextInput` without adding custom overrides. These themes are filtered out from the ["all tokens" story](https://ui-ivzahz34a-commercetools.vercel.app/?path=/story/basics-tokens--all-tokens).

It seems this approach doesn't cause any breaking changes in the `docs-kit`. This is the [deployment link](https://harmonious-stroopwafel-158412.netlify.app/) of the root folder of docs website with `@commercetools-uikit/design-system` package within `@commercetools-docs/ui-kit` built from this PRs codebase:
`"@commercetools-uikit/design-system": "https://uikit-theming.netlify.app/commercetools-uikit-design-system-15.2.0.tgz"`. There is no `<ThemeProvider>` added and all css vars fall back to fixed values.

Reply:
I updated the PR with changes that we discussed:
- [custom theme overrides in `<ThemeProvider>`](https://github.com/commercetools/ui-kit/blob/bf1c8698cc73948f67d409038b7291a3855ac5af/design-system/src/theme-provider.tsx#L81:L84) + [inclued that in the vrt](https://github.com/commercetools/ui-kit/blob/bf1c8698cc73948f67d409038b7291a3855ac5af/design-system/src/theme-provider.visualroute.jsx#L64:L91)
- [passing a ref to `<ThemeProvider>` to make it scoped locally](https://github.com/commercetools/ui-kit/blob/bf1c8698cc73948f67d409038b7291a3855ac5af/design-system/src/theme-provider.tsx#L77:L79) - I used that approach instead of the `getDefaultParentSelector` prop but if you prefer that please let me know
- in vrts removed fragments using local custom themes (applied to Tag, Link, TextInput). I left the [dark theme](test/percy/local-theme-provider.jsx) in vrts though